### ISSUE TITLE
build before run on debug restart

### ISSUE THREAD

**Issue by dvirtz** [microsoft/vscode-cmake-tools#1604]
### Brief Issue Summary

having `Build Before Run` and changing code during a debug session doesn't cause the project to be built when debug restart button is pressed, enforcing me to stop debug and start it again.

### Expected:

1. Start debug
2. Change some code
3. Push Debug:Restart button
4. Project is built and then run

### Apparent Behavior:

1. Start debug
2. Change some code
3. Push Debug:Restart button
4. Debug restarts without build
5. Stop debug
6. Start debug
7. Project is built and then run

### Platform and Versions

- **Operating System**: macOS
- **CMake Version**: 3.18.0
- **VSCode Version**: 1.52.1
- **CMake Tools Extension Version**: 1.5.3
- **Compiler/Toolchain**: Clang 8.0.1

**Comment by bobbrow** [comment]
We'd have to check, but I don't think we get a notification from VS Code when that button is pressed.  This is potentially a feature request for VS Code.

**Comment by dvirtz** [comment]
Opened https://github.com/microsoft/vscode/issues/114664

**Comment by dvirtz** [comment]
@gcampbell-msft was this fixed? I don't it being mentioned in the change log

**Comment by gcampbell-msft** [comment]
No, it seems that the conversation here: https://github.com/microsoft/vscode/issues/114664, decided that this wouldn't be supported from VSCode
